inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp

```diff
@@ -235,7 +235,132 @@ ie::CNNNetwork FrontEnd::convertNetwork(ie::CNNNetwork& network) {
     IE_SUPPRESS_DEPRECATED_END
 }
 
-std::set<std::string> FrontEnd::checkSupportedLayers(const ie::CNNNetwork& network) {
+std::vector<ie::CNNNetwork> FrontEnd::checkSupportedNetworks(const ie::CNNNetwork& network, std::set<std::string>& namesToExclude) {
+    std::vector<ie::CNNNetwork> supportedNetworks;
+    if (network.getFunction() && network.getFunction()->is_dynamic()) {
+        auto copyNetwork = InferenceEngine::details::cloneNetwork(network);
+        try {
+            convertNetwork(copyNetwork);
+        } catch(vpu::details::VPUException e) {
+            std::string name = e.what();
+            std::string nameDelimiter("node ");
+            name.erase(0, name.find(nameDelimiter) + nameDelimiter.length());
+            nameDelimiter = " of";
+            name.erase(name.find(nameDelimiter), name.length());
+            auto function = network.getFunction();
+            bool notFound = true;
+            ov::ParameterVector paramsTop;
+            ov::NodeVector resultsLow;
+            ov::NodeVector networkTop;
+            ov::NodeVector networkLow;
+            for (auto& node : function->get_ordered_ops()) {
+                if (node->get_friendly_name() == name) {
+                    notFound = false;
+                    continue;
+                }
+                if (notFound) {
+                    networkTop.push_back(node);
+                    continue;
+                }
+                networkLow.push_back(node);
+            }
+            if (notFound) {
+                throw e;
+            }
+            namesToExclude.insert(name);
+            if (networkTop.size() != 0) {
+                ov::NodeVector resultsTop;
+                std::map<std::string, std::shared_ptr<ov::Node>> newNodes;
+                for (auto& node : networkTop) {
+                    if (strcmp(node->get_type_info().name, "Parameter") == 0) {
+                        auto input = std::dynamic_pointer_cast<ngraph::opset3::Parameter>(node);
+                        auto newNode = std::make_shared<ngraph::opset3::Parameter>(input->get_element_type(), input->get_partial_shape());;
+                        paramsTop.push_back(newNode);
+                        continue;
+                    }
+                    ov::OutputVector inputs;
+                    for (auto& input : node->input_values()) {
+                        auto& newNode = newNodes[input.get_node()->get_friendly_name()];
```

Comment:
But you haven't changed `newNodes` until now. Is it correct to use it here?
Isn't `newNodes.count(input.get_node()->get_friendly_name()) == 0` here?

Reply:
There is a mistake, every node that's created in this cycle should be added to `newNodes`, I'll fix it.

Reply:
I've fixed it